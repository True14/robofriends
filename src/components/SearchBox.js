import React, { memo } from "react";

const SearchBox = ({ searchChange }) => {
  return (
    <div className="pa2">
      <input
        aria-label="Search Robots"
        id="search-box"
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        onChange={searchChange}
      />
    </div>
  );
};

export default memo(SearchBox);
