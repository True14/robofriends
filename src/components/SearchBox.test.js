import React from "react";
import { shallow } from "enzyme";
import SearchBox from "./SearchBox";

describe("The SearchBox component", () => {
  let wrapper;
  let searchChange;
  beforeEach(() => {
    searchChange = jest.fn();
    wrapper = shallow(<SearchBox searchChange={searchChange} />);
  });

  it("renders withouth crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a input field with id 'search-box", () => {
    expect(wrapper.find("input#search-box")).toHaveLength(1);
  });

  it("calls searchChange from props when input field is changed", () => {
    wrapper
      .find("input#search-box")
      .simulate("change", { target: { value: "john" } });
    expect(searchChange).toHaveBeenCalledTimes(1);
  });
});
