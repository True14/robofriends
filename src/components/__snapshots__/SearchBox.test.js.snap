// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SearchBox component renders withouth crashing 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="Search Robots"
    className="pa3 ba b--green bg-lightest-blue"
    id="search-box"
    onChange={[MockFunction]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
